src/plugin/file/kml/mime.js

```diff
@@ -39,7 +41,38 @@ os.file.mime.register(
  */
 plugin.file.kml.mime.KMZ_TYPE = 'application/vnd.google-earth.kmz';
 
+
+/**
+ * Determine if this file is a KMZ file.  Currently, the logic is:
+ * Must contain *.kml file(s) and have a *.kmz filename
+ *
+ * @param {ArrayBuffer} buffer
+ * @param {os.file.File=} opt_file
+ * @param {*=} opt_context
+ * @return {!goog.Promise<*|undefined>}
+ */
+plugin.file.kml.detectKmz = function(buffer, opt_file, opt_context) {
+  var retVal;
+  var kmzRegex = /\.kmz$/i;
+  var kmlRegex = /\.kml$/i;
```

Comment:
I am generally leery of filename tests for typing files. They tend to bite us with generate files as the naming schemes are often meant for machine to machine communication, so they leave off file extensions in the names. This might not technically be a valid thing for a KMZ to do, but if we can apply more robust checking of the zip file entries (i.e. our existing KML mime check which looks at the internal element names/namespaces).

Reply:
The case here is that it is difficult to determine whether a zip full of KML files is a single KMZ with multiple KML files (a single import) or an actual user-zipped set of several KMLs (zip multi import). The best distinction between those two is the file extension.

Reply:
Added the TYPE check.  Now I feel much more comfortable about this too, and it still passes the KML and KMZ mime.test.js tests.